cli/flox/src/commands/services/stop.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::providers::services::{stop_services, ProcessStates};
+use tracing::instrument;
+
+use crate::commands::{environment_select, EnvironmentSelect};
+use crate::subcommand_metric;
+use crate::utils::message;
+
+#[derive(Bpaf, Debug, Clone)]
+pub struct Stop {
+    #[bpaf(external(environment_select), fallback(Default::default()))]
+    environment: EnvironmentSelect,
+
+    /// Names of the services to stop
+    #[bpaf(positional("name"))]
+    names: Vec<String>,
+}
+
+impl Stop {
+    #[instrument(name = "stop", skip_all)]
+    pub async fn handle(self, flox: Flox) -> Result<()> {
+        subcommand_metric!("services::stop");
+
+        let env = self
+            .environment
+            .detect_concrete_environment(&flox, "Services in")?
+            .into_dyn_environment();
```

Comment:
blocking: the ticket says to error if not in an activation, although I kinda think what you have here is better. If we are going to leave it this way, at least in followup I think we probably need to change the error message for an environment that isn't activated or doesn't have services from:
```
❌ ERROR: couldn't connect to service manager
```
to
```
❌ ERROR: environment doesn't have any running services
```

Reply:
I think it makes sense to allow stopping services from outside an activation. I don't remember a discussion about why we would disallow that. Starting services is a different matter because it's not clear who would clean up services.

I'm not sure about changing that error message. The current error message is broad, but at least accurate, and I don't know that being in the presence of an activation should give you different information.

Reply:
Note that calling `flox services stop` from anywhere (inside or outside an activation) doesn't actually shut down `process-compose` because of our dummy service.

Reply:
> Starting services is a different matter because it's not clear who would clean up services.

We could check in the env registry if there's already a watchdog. If so, start. If not, say sorry there aren't any activations (this would mean any activate would need to start a watchdog, even if it didn't have -s passed)

Reply:
Sure, we could do that. I'm a little uncomfortable that we're building a distributed daemon operating on JSON-as-a-database, but we talked in the service management meeting that we will likely have a discussion about a daemon after launch.

Reply:
Okay I'm going to append to the ticket saying `stop` doesn't have to be in an activation. We can deal with error polish later